contracts/RNGBlockhash.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.6.6;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/SafeCast.sol";
+
+import "./RNGInterface.sol";
+
+contract RNGBlockhash is RNGInterface, Ownable {
+  using SafeMath for uint256;
+  using SafeCast for uint256;
+
+  /// @dev A counter for the number of requests made used for request ids
+  uint32 internal requestCount;
+
+  /// @dev A list of random numbers from past requests mapped by request id
+  mapping(uint32 => uint256) internal randomNumbers;
+
+  /// @notice Public constructor
+  constructor() public { }
+
+  /// @notice Gets the last request id used by the RNG service
+  /// @return requestId The last request id used in the last request
+  function getLastRequestId() external override view returns (uint32 requestId) {
+    return requestCount;
+  }
+
+  /// @notice Gets the Fee for making a Request against an RNG service
+  /// @return feeToken The address of the token that is used to pay fees
+  /// @return requestFee The fee required to be paid to make a request
+  function getRequestFee() external override view returns (address feeToken, uint256 requestFee) {
+    return (address(0), 0);
+  }
+
+  /// @notice Sends a request for a random number to the 3rd-party service
+  /// @dev Some services will complete the request immediately, others may have a time-delay
+  /// @dev Some services require payment in the form of a token, such as $LINK for Chainlink VRF
+  /// @return requestId The ID of the request used to get the results of the RNG service
+  /// @return lockBlock The block number at which the RNG service will start generating time-delayed randomness.  The calling contract
+  /// should "lock" all activity until the result is available via the `requestId`
+  function requestRandomNumber() external virtual override returns (uint32 requestId, uint32 lockBlock) {
+    lockBlock = uint32(block.number);
+    requestId = _generateRandomness(_getSeed());
+
+    emit RandomNumberRequested(requestId, msg.sender);
+  }
+
+  /// @notice Checks if the request for randomness from the 3rd-party service has completed
+  /// @dev For time-delayed requests, this function is used to check/confirm completion
+  /// @param requestId The ID of the request used to get the results of the RNG service
+  /// @return isCompleted True if the request has completed and a random number is available, false otherwise
+  function isRequestComplete(uint32 requestId) external virtual override view returns (bool isCompleted) {
+    return randomNumbers[requestId] > 0;
```

Comment:
Here, the request will be complete once a blockhash is available for a block *after* the lock block